### ISSUE TITLE
Run all bazel tests in CI?

### ISSUE THREAD

**Issue by mostafaelhoushi** [facebookresearch/CompilerGym#693]
Can we add to the CI:
- `bazel test //examples/...` to run all bazel tests in the `examples` directory
- or even `bazel test ...` to run all bazel tests in the repo

**Comment by mostafaelhoushi** [comment]
@ChrisCummins can we do that? Do you know where in the ci.yaml file or Makefile we can add the `bazel test` command?

This will make it easier to enable example tests in CI.... then in the future we can enable examples tests without bazel in CI

**Comment by ChrisCummins** [comment]
> @ChrisCummins can we do that? Do you know where in the ci.yaml file or Makefile we can add the `bazel test` command?

Yes, it is totally do-able. I removed `bazel test` jobs in https://github.com/facebookresearch/CompilerGym/pull/392. It would be worth reading through the PR message and https://github.com/facebookresearch/CompilerGym/issues/385 for the rationale.

Would it be possible to use the [build-linux-cmake](https://github.com/facebookresearch/CompilerGym/blob/development/.github/workflows/ci.yaml#L43) job to compile any missing C++ tests for execution by cmake? I see that we compile with `COMPILER_GYM_BUILD_TESTS=ON`, but then [don't actually use them](https://github.com/facebookresearch/CompilerGym/blob/development/.github/workflows/ci.yaml#L237-L238).

Cheers,
Chris
